### PR TITLE
Closes #17325. Add view detection for MySQL.

### LINKED ISSUE #17325: Command migrate/fresh fails on DB view
Console command migrate/fresh doesn't work as expected.
It fails on deletion of DB View. 

MySQL returns following error message:
`#1051 - Unknown table 'db-name.view-name'`
which is not covered by current fix from  #16278

Yii2 - 2.0.19-dev
MySQL - Percona 5.7

### PR DESCRIPTION
**Pull request by alexkart**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #17325

### CODE REVIEW

File: framework/console/controllers/MigrateController.php [hunk 1]
```diff
@@ -324,6 +324,27 @@ protected function truncateDatabase()
         }
     }
 
+    /**
+     * Determines whether the error message related to deleting a view or not
```
Comment (samdark) [inline framework/console/controllers/MigrateController.php]:
```suggestion
     * Determines whether the error message is related to deleting a view or not
```

File: framework/console/controllers/MigrateController.php [hunk 2]
```diff
@@ -324,6 +324,27 @@ protected function truncateDatabase()
...
+     * @param $errorMessage
+     * @return bool
+     */
+    private function isView($errorMessage)
```
Comment (samdark) [inline framework/console/controllers/MigrateController.php]:
```suggestion
    private function isViewRelated($errorMessage)
```

File: framework/console/controllers/MigrateController.php [hunk 3]
```diff
@@ -314,7 +314,7 @@ protected function truncateDatabase()
                 $db->createCommand()->dropTable($schema->name)->execute();
                 $this->stdout("Table {$schema->name} dropped.\n");
             } catch (\Exception $e) {
-                if (strpos($e->getMessage(), 'DROP VIEW to delete view') !== false) {
+                if ($this->isView($e->getMessage())) {
```
Comment (samdark) [inline framework/console/controllers/MigrateController.php]:
```suggestion
                if ($this->isViewRelated($e->getMessage())) {
```

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by alexkart** [comment]
> Thank you for putting effort in the improvement of the Yii framework.
> We have reviewed your pull request.
> 
> In order for the framework and your solution to remain stable in the future, we have a unit test requirement in place. Therefore we can only accept your pull request if it is covered by unit tests.
> 
> Could you add these please?
> 
> Thanks!
> 
> P.S. If you have any questions about the creation of unit tests? Don't hesitate to ask for support. [More information about unit tests](http://www.yiiframework.com/doc-2.0/guide-test-unit.html)
> 
> _This is an automated comment, triggered by adding the label `pr:request for unit tests`._

Is it possible to run tests for MySQL?

**Comment by alexkart** [comment]
Have I done it correctly?

**Comment by samdark** [comment]
Yes, it is possible to run MySQL. You did it but there are connection configs and it is better to use these and not hardcoding credentials.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thank you!

**Pull request merged**